app/models/question.rb

```diff
@@ -2,9 +2,25 @@ class Question < ApplicationRecord
   include RailsAdminConf
 
   has_many :answers, dependent: :destroy, inverse_of: :question # the questions choices
+  has_many :user_questions_answers, dependent: :destroy
 
   accepts_nested_attributes_for :answers, allow_destroy: true
 
   validates :name, presence: true, uniqueness: true
+  validates_presence_of :category
+
+  enum category: [:daily, :weekly]
+
+  # It will filter the questions based on activeness and category i.e daily and weekly
+  scope :active, -> (category) { where(active: true, category: category) }
+
+  # It will only fetch those questions of user which has not been answered
+  scope :not_seen, lambda { |user_id|
```

Comment:
Can we perf test this ? Say for a million entries in `questions` table and about 10K entries in the `user_questions_answers` table ?

Reply:
What would you like to test exactly?

Reply:
This scope is called everytime a user requests for a daily/weekly question. So please try to insert some dummy records and check the response time of the `get questions` api.